src/hotspot/share/nmt/virtualMemoryTracker.hpp

```diff
@@ -372,40 +366,62 @@ class VirtualMemoryWalker : public StackObj {
    virtual bool do_allocation_site(const ReservedMemoryRegion* rgn) { return false; }
 };
 
-// Main class called from MemTracker to track virtual memory allocations, commits and releases.
-class VirtualMemoryTracker : AllStatic {
-  friend class VirtualMemoryTrackerTest;
-  friend class CommittedVirtualMemoryTest;
+class RegionsTree;
+class VirtualMemoryTracker {
+ private:
+  RegionsTree *_tree;
 
  public:
-  static bool initialize(NMT_TrackingLevel level);
-
-  static bool add_reserved_region (address base_addr, size_t size, const NativeCallStack& stack, MemTag mem_tag = mtNone);
+  VirtualMemoryTracker(bool is_detailed_mode);
 
-  static bool add_committed_region      (address base_addr, size_t size, const NativeCallStack& stack);
-  static bool remove_uncommitted_region (address base_addr, size_t size);
-  static bool remove_released_region    (address base_addr, size_t size);
-  static bool remove_released_region    (ReservedMemoryRegion* rgn);
-  static void set_reserved_region_type  (address addr, MemTag mem_tag);
+  bool add_reserved_region       (address base_addr, size_t size, const NativeCallStack& stack, MemTag mem_tag = mtNone);
+  bool add_committed_region      (address base_addr, size_t size, const NativeCallStack& stack);
+  bool remove_uncommitted_region (address base_addr, size_t size);
+  bool remove_released_region    (address base_addr, size_t size);
+  bool remove_released_region    (ReservedMemoryRegion* rgn);
+  void set_reserved_region_tag   (address addr, MemTag mem_tag);
 
   // Given an existing memory mapping registered with NMT, split the mapping in
   //  two. The newly created two mappings will be registered under the call
-  //  stack and the memory tag of the original section.
-  static bool split_reserved_region(address addr, size_t size, size_t split, MemTag mem_tag, MemTag split_type);
+  //  stack and the memory tags of the original section.
+  bool split_reserved_region(address addr, size_t size, size_t split, MemTag mem_tag, MemTag split_mem_tag);
 
   // Walk virtual memory data structure for creating baseline, etc.
-  static bool walk_virtual_memory(VirtualMemoryWalker* walker);
+  bool walk_virtual_memory(VirtualMemoryWalker* walker);
 
   // If p is contained within a known memory region, print information about it to the
   // given stream and return true; false otherwise.
-  static bool print_containing_region(const void* p, outputStream* st);
+  bool print_containing_region(const void* p, outputStream* st);
 
   // Snapshot current thread stacks
-  static void snapshot_thread_stacks();
-
- private:
-  static SortedLinkedList<ReservedMemoryRegion, compare_reserved_region_base>* _reserved_regions;
+  void snapshot_thread_stacks();
+  void apply_summary_diff(VMATree::SummaryDiff diff);
+  RegionsTree* tree() { return _tree; }
+
+  class Instance : public AllStatic {
+    friend class VirtualMemoryTrackerTest;
+    friend class CommittedVirtualMemoryTest;
+
```

Comment:
These two classes doesn't exist anymore.

Reply:
The first got back to life. The second is removed.